### PR TITLE
NFTBank Refactor and Integration Tests

### PR DESCRIPTION
**Pull request by jnapprogs**
NFTBank refactorings and integration tests.

### CODE REVIEW

**Comment by jnapprogs** [comment]
Here is the link to the old PR.  This contains the reviews that this PR is reflected upon.

https://github.com/linkpoolio/external-adapters-js/pull/96

File: packages/sources/nftbank/src/endpoint/estimate.ts [hunk 1]
```diff
@@ -1,59 +1,100 @@
...
 
-export const supportedEndpoints = ['nftbank']
+import type { ResponseSchema } from '../types'
+import { NftCollection, PricingAsset } from '../const'
+import { getEstimateFor } from '../utils'
 
-export const endpointResultPaths = {
-  nftbank: 'nftbank',
-}
+export const supportedEndpoints = ['nftbank']
```
Comment (vnavascues) [inline packages/sources/nftbank/src/endpoint/estimate.ts]: Replace the endpoint name for `estimate`:

```typescript
export const supportedEndpoints = ['estimate']
```
1. It is more meaningful than `nftbank`.
2. Aligned with the file name `estimate.ts`.
3. It is the endpoint name you've been using on the tests.
4. In our 2-3 TOML jobspecs is the endpoint name we use (it happens that this EA has a default endpoint logic).

```toml
    nftbank                         [type=bridge
                                     name="nftbank"
                                     requestData="{\\"data\\": {\\"endpoint\\": \\"estimate\\", \\"nftCollection\\": $(decode_cbor.nftCollection), \\"nftId\\": $(decode_cbor.nftId), \\"pricingAsset\\": $(decode_cbor.pricingAsset)}}"]
```

File: packages/sources/nftbank/test/integration/__snapshots__/adapter.test.ts.snap [hunk 2]
```diff
@@ -0,0 +1,33 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`execute estimate endpoint success calls returns the proper result 1`] = `
```
Comment (vnavascues) [inline packages/sources/nftbank/test/integration/__snapshots__/adapter.test.ts.snap:3]: As you can see, now your EA return can preserve the original API response (it includes `...response.data`). By running the integration test with `API_VERBOSE=true` you are including the genuine API response data (which comes from your fixture) :)

**Review by vnavascues — CHANGES_REQUESTED** [review]
Almost there! As far as I'm aware just these 2 things:
- Amend the `package.json` version to `1.0.25`.
- Amend the `supportedEndpoints` endpoint name in `estimate.ts`, from `nftbank` to `estimate`.

File: packages/sources/nftbank/src/endpoint/estimate.ts (hunk 1 above)
Comment (jnapprogs) [reply packages/sources/nftbank/src/endpoint/estimate.ts]: Sounds good!  Will make the adjustments now

Comment (jnapprogs) [reply packages/sources/nftbank/src/endpoint/estimate.ts]: Changes have been applied

File: packages/sources/nftbank/test/integration/__snapshots__/adapter.test.ts.snap (hunk 2 above)
Comment (jnapprogs) [reply packages/sources/nftbank/test/integration/__snapshots__/adapter.test.ts.snap:3]: That is crazy...

Comment (jnapprogs) [reply packages/sources/nftbank/test/integration/__snapshots__/adapter.test.ts.snap:3]: In a good way

**Pull request closed without merging**